### PR TITLE
Brings in Stefan's correlation code

### PR DESCRIPTION
**Pull request by jwdebelius**

### CODE REVIEW

**Comment by jwdebelius** [comment]
@wasade, could you please review and/or tag Stefan?

Thanks!

File: ipynb/Metadata cross correlation.ipynb [hunk 1]
```diff
@@ -967,6 +967,72 @@
     "dm = skbio.DistanceMatrix(data=(1 - dm_corr1.values), ids=dm_corr1.index)\n",
     "dm.write('./correlations_dm.txt')"
    ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Finally, we'll perform higherarchical clustering on the distance matrix, to determine the groups."
```
Comment (wasade) [inline ipynb/Metadata cross correlation.ipynb]: hierarchical

File: ipynb/Metadata cross correlation.ipynb [hunk 2]
```diff
@@ -967,6 +967,72 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "threshold = 0.29\n",
```
Comment (wasade) [inline ipynb/Metadata cross correlation.ipynb:1004]: can you add: `# chosen as it approximated a cluster width of correlation of 0.5`

and please reword if that isn't as cleanly stated as it should be

File: ipynb/Metadata cross correlation.ipynb [hunk 3]
```diff
@@ -967,6 +967,72 @@
...
+    "format_long = True\n",
+    "\n",
+    "seen_clusters = dict()\n",
+    "f = open('agp_metadata_clusters_%f.txt' % threshold, 'w')\n",
+    "f.write('# threshold = %f, ward clustering of %s matrix\\n' % (threshold, str(dm_.shape)))\n",
+    "for i, tip in enumerate(tree_.tips()):\n",
+    "    curr_node = tip\n",
+    "    tip_dist = tip.length\n",
+    "    while tip_dist <= 0.29:\n",
```
Comment (wasade) [inline ipynb/Metadata cross correlation.ipynb]: can this use the `threshold` variable?

File: ipynb/Metadata cross correlation.ipynb [hunk 4]
```diff
@@ -967,6 +967,72 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "tree_ = skbio.tree.TreeNode.from_linkage_matrix(ward(dm_.condensed_form()), dm_.ids)"
```
Comment (wasade) [inline ipynb/Metadata cross correlation.ipynb]: should this be `dm` instead of `dm_`?

**Comment by jwdebelius** [comment]
@wasade: Addressed, I think

**Pull request merged**